### PR TITLE
Android: Quality of life improvement for android cutout handling.

### PR DESCRIPTION
**Pull request by marcosantos98**
## "Problem":

Currently, the developer is required to know the Android API to get real fullscreen in Android devices with cutouts. This was already worked on before by @MrStahlfelge in #5714 when `Graphics::getSafeInsetsXX` was introduced. He also provided an example of how to get fullscreen on screens with cutouts.

https://github.com/libgdx/libgdx/blob/fd535ab38900cd266631a7bbeab83d10c1bc5403/tests/gdx-tests-android/src/com/badlogic/gdx/tests/android/GdxTestActivity.java#L35C1-L39C4

## Improvement:

- Add a config option called `AndroidApplicationConfiguration::renderUnderCutout` that allows the end user to use the space under the cutout.
- Check if the API is available on the device and use it if `AndroidApplicationConfiguration::renderUnderCutout` is set to true.

## Notes:

Not tested on IOS, licensing to expensive for a simple project. 

## Example:
```java
import android.os.Bundle;

import com.badlogic.gdx.backends.android.AndroidApplication;
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration;

public class GdxTestActivity extends AndroidApplication {

	public void onCreate (Bundle bundle) {
		super.onCreate(bundle);

		AndroidApplicationConfiguration config = new AndroidApplicationConfiguration();
		config.renderUnderCutout = true;
		initialize(new Game(), config);
	}
}

```

### CODE REVIEW

**Comment by Frosty-J** [comment]
Based on the screenshots I've seen, iOS already seems to render in the cutout area, so I don't think anything needs doing for it. Thank you for providing this - it'll be easier than me pointing people to the cutouts documentation and telling them to use short edges without actually explaining how because I never bothered figuring it out for myself.

File: backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java [hunk 1]
```diff
@@ -186,6 +191,15 @@ protected void createWakeLock (boolean use) {
 		}
 	}
 
+	@TargetApi(28)
```
Comment (obigu) [inline backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java:194]: Maybe use `Build.VERSION_CODES.P` here as well to see they match?

File: backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java [hunk 2]
```diff
@@ -186,6 +191,15 @@ protected void createWakeLock (boolean use) {
...
+	private void tryRenderUnderCutout(boolean render) {
+		if(render && getVersion() >= Build.VERSION_CODES.P) {
+			WindowManager.LayoutParams lp = getWindow().getAttributes();
+			lp.layoutInDisplayCutoutMode = WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES;
+			getWindow().setAttributes(lp);
```
Comment (obigu) [inline backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java:199]: Haven't tested but do we need this last line?

**Review by obigu — COMMENTED** [review]
Thanks for the contribution! Looks good, just 2 minor things. I still haven't tested properly though, will update when I do.

File: backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java (hunk 1 above)
Comment (marcosantos98) [reply backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java:194]: I followed the used format in `useImmersiveMode`.

File: backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java (hunk 2 above)
Comment (marcosantos98) [reply backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java:199]: You are right, I normally use it with the line

File: backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java [hunk 3]
```diff
@@ -186,6 +191,14 @@ protected void createWakeLock (boolean use) {
 		}
 	}
 
+	@TargetApi(Build.VERSION_CODES.P)
+	private void tryRenderUnderCutout(boolean render) {
```
Comment (obigu) [inline backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java:195]: I've tested the PR using libGDX tests and it seems to be working. 

Just a final change (sorry for being nitpicky). Can you change the name of the method `tryRenderUnderCutout` to something more descriptive such as setLayoutInDisplayCutoutMode(boolean)`.

Comment (marcosantos98) [reply backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java]: No problem. Sorry for the late response

**Review by obigu — APPROVED** [review]
Thanks!

**Comment by tommyettinger** [comment]
It looks like you just need to run `gradlew spotlessApply` in the repo for it to be easier to merge. I think the other parts of automated testing were all perfectly fine.

**Comment by obigu** [comment]
I've pushed changes to merge and run Spotless and it's fixed now (we had issues with the build at the same time this PR was created, that's why it's been messy).

Do you mind adding a line to CHANGES and we're good to merge?

**Comment by marcosantos98** [comment]
> I've pushed changes to merge and run Spotless and it's fixed now (we had issues with the build at the same time this PR was created, that's why it's been messy).
> 
> Do you mind adding a line to CHANGES and we're good to merge?

Done

**Pull request merged**